### PR TITLE
More thread stuff

### PR DESCRIPTION
**Pull request by itohatweb**

### CODE REVIEW

**Comment by ayntee** [comment]
@itohatweb is this PR ready for review?

File: src/handlers/channels/THREAD_LIST_SYNC.ts [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-      return discordenoChannel;
+      return threadData;
     })
   );
 
-  const threads = new Collection<bigint, DiscordenoChannel>(discordenoChannels.map((t) => [t.id, t]));
-
-  eventHandlers.threadListSync?.(threads, payload.members, snowflakeToBigint(payload.guildId));
+  eventHandlers.threadListSync?.(
+    new Collection(threads.map((t) => [t.id, t])),
```
Comment (itohatweb) [inline src/handlers/channels/THREAD_LIST_SYNC.ts:23]: convert the thread

File: src/structures/channel.ts [hunk 2]
```diff
@@ -110,6 +110,8 @@ export async function createDiscordenoChannel(data: Channel, guildId?: bigint) {
   (Object.keys(rest) as (keyof typeof rest)[]).forEach((key) => {
     eventHandlers.debug?.("loop", `Running forEach loop in createDiscordenoChannel function.`);
 
+    if (key === "threadMetadata") return;
+
```
Comment (itohatweb) [inline src/structures/channel.ts:114]: Not needed? cause we will never create thread channels with this

File: src/types/channels/threads/thread_member.ts [hunk 3]
```diff
@@ -1,10 +1,22 @@
-export interface ThreadMember {
```
Comment (itohatweb) [inline src/types/channels/threads/thread_member.ts:1]: I would say use Generics like I showed in deno lib server

**Pull request merged**